graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/parser/ActionParser.java

```diff
@@ -47,32 +46,42 @@ String generateAction(RuleBuilderStep step, boolean generateSimulatorFields) {
             throw new IllegalArgumentException("Function " + step.function() + " not available as action for rule builder.");
         }
 
-        FunctionDescriptor<?> function = actions.get(step.function()).descriptor();
+        final RuleFragment ruleFragment = actions.get(step.function());
+        FunctionDescriptor<?> function = ruleFragment.descriptor();
+
         String syntax = "  ";
         if (Objects.nonNull(step.outputvariable())) {
```

Comment:
We could create a set of validators that we need to run and separate that out of the parsing, what do you think?

Reply:
yes, good idea. we need to pull out validation.